From: Joshua Sager <joshua.sager@shopify.com>
Date: Mon, 13 Sep 2021 17:50:33 -0400
Subject: [PATCH] Initialize rows using heap for large images

--- a/libimagequant.c
+++ b/libimagequant.c
@@ -1649,7 +1649,7 @@ LIQ_EXPORT LIQ_NONNULL liq_error liq_write_remapped_image(liq_result *result, li
         return LIQ_BUFFER_TOO_SMALL;
     }
 
-    unsigned char *rows[input_image->height];
+    unsigned char **rows = malloc(input_image->height * sizeof(unsigned char *));
     unsigned char *buffer_bytes = buffer;
     for(unsigned int i=0; i < input_image->height; i++) {
         rows[i] = &buffer_bytes[input_image->width * i];
