From: Joshua Sager <joshua.sager@shopify.com>
Date: Mon, 13 Sep 2021 17:57:44 -0400
Subject: [PATCH] free rows after remapping

--- a/libimagequant.c
+++ b/libimagequant.c
@@ -1654,7 +1654,10 @@ LIQ_EXPORT LIQ_NONNULL liq_error liq_write_remapped_image(liq_result *result, li
     for(unsigned int i=0; i < input_image->height; i++) {
         rows[i] = &buffer_bytes[input_image->width * i];
     }
-    return liq_write_remapped_image_rows(result, input_image, rows);
+    
+    liq_error err = liq_write_remapped_image_rows(result, input_image, rows);
+    free(rows);
+    return err;
 }
 
 LIQ_EXPORT liq_error liq_write_remapped_image_rows(liq_result *quant, liq_image *input_image, unsigned char **row_pointers)
